switchlink/switchlink_main.c

```diff
@@ -288,9 +289,18 @@ static void nl_process_event_loop(void) {
 
     ret = select(num_fds, &read_fds, NULL, NULL, NULL);
     if (ret == -1) {
-      perror("pselect");
+      char errbuf[64];
+
+      // errno is volatile. Make a local copy of its value.
+      int err = errno;
+
+      // This is not really an error condition. Ignore it.
```

Comment:
`(errno == EINTR)` should be `(err == EINTR)`.

The idea is to make a local copy of errno and then use the copy for _all_ subsequent references, to ensure complete consistency. `errno` is a volatile variable, which means that its state can be changed by another thread at any time.